ci-operator/step-registry/ipi/conf/azurestack/ipi-conf-azurestack-commands.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+
+REGION="ppe3"
+echo "Azure region: ${REGION}"
+
+workers=3
+if [[ "${SIZE_VARIANT}" == "compact" ]]; then
+  workers=0
+fi
+master_type=null
+if [[ "${SIZE_VARIANT}" == "xlarge" ]]; then
+  master_type=Standard_D32s_v3
+elif [[ "${SIZE_VARIANT}" == "large" ]]; then
+  master_type=Standard_D16s_v3
+elif [[ "${SIZE_VARIANT}" == "compact" ]]; then
+  master_type=Standard_D8s_v3
+fi
+
+ENDPOINT="https://management.ppe3.stackpoc.com"
+echo "ASH ARM Endpoint: ${ENDPOINT}"
+
+azurestackjson=$(cat <<EOF
+{
+  "name": "AzureStackCloud",
+  "resourceManagerEndpoint": "https://management.ppe3.stackpoc.com",
+  "activeDirectoryEndpoint": "https://login.microsoftonline.com/",
+  "galleryEndpoint": "https://providers.ppe3.local:30016/",
+  "storageEndpointSuffix": "https://providers.ppe3.local:30016/",
+  "serviceManagementEndpoint": "https://management.stackpoc.com/81c9b804-ec9e-4b5a-8845-1d197268b1f5",
+  "graphEndpoint":                "https://graph.windows.net/",
+  "resourceIdentifiers": {
+    "graph": "https://graph.windows.net/"
+  }
+}
+EOF
+)
+
+echo $azurestackjson >> "${SHARED_DIR}/azurestackcloud.json"
+
+export AZURE_ENVIRONMENT_FILEPATH=${SHARED_DIR}/azurestackcloud.json
+export OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE=quay.io/padillon/ash-release-mirror:latest
+export OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE=https://rhcossa.blob.ppe3.stackpoc.com/vhd/rhcos4.vhd
+cat "/var/run/azurestack-cluster-secrets/service-principal" >> "${SHARED_DIR}/osServicePrincipal.json"
```

Comment:
I think the [most recent failures in CI](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/18796/rehearse-18796-pull-ci-openshift-installer-master-e2e-azurestack/1417516485978361856#1:build-log.txt%3A53) are due to the fact that we're not setting the auth creds correctly:
```
level=info msg=Credentials loaded from file "/var/run/secrets/ci.openshift.io/cluster-profile/osServicePrincipal.json"
level=fatal msg=failed to fetch Master Machines: failed to load asset "Install Config": [platform.azure.region: Invalid value: "ppe3": region "ppe3" is not valid or not available for this account, controlPlane.platform.azure.type: Invalid value: "Standard_D4s_v3": not found in region ppe3, compute[0].platform.azure.type: Invalid value: "Standard_D4s_v3": not found in region ppe3]
```
It looks like this is using the `osServicePrincipal.json` for public Azure--not Azure Stack. Do we know where this is/should be set?

Reply:
`osServicePrincipal.json` comes from the vault here: https://vault.ci.openshift.org/ui/vault/secrets

I added the secrets there manually by creating a service Principal for CI and synced it to the cluster storage. I don't think that's the problem here, the failure was due to the Azure SDK version used. That run was done before your IPI PR and I'm hoping it's fixed. Failures now are mostly quota problems.

Reply:
The retrieval from the vault is working, but they're not being set correctly after that. In UPI they are set explicitly: 
```
export AZURE_AUTH_LOCATION="${SHARED_DIR}/osServicePrincipal.json"
```
This is missing from the IPI steps. It is fine to add this for testing, but it would be good if we did not have to set this at all in the steps, but rather could rely on the cluster profile. I don't see this set in the plain azure steps, but I could be missing something.

Reply:
This shouldn't depend on the IPI PR, but should have been resolved by openshift/installer#5024 which merged a while ago. It doesn't look like this error shows up in UPI

Reply:
>This is missing from the IPI steps
I'm actually trying to test at least one of these to see if the env setup works. Thought UPI would be easier.

> I don't see this set in the plain azure steps, but I could be missing something.
It's actually here in the ipi-install-install : azure4) export AZURE_AUTH_LOCATION=${CLUSTER_PROFILE_DIR}/osServicePrincipal.json;; I'll add it here.

>This shouldn't depend on the IPI PR, but should have been resolved by openshift/installer#5024 which merged a while ago. It doesn't look like this error shows up in UPI
I think it was not getting the region since we did not use the hybrid profile until IPI PR added it to the vendor modules.

Reply:
> It's actually here in the ipi-install-install :
> 
> https://github.com/openshift/release/blob/2148a6130def51de1e31270f354790596338b7d2/ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh#L74
> 
> I'll add it here.

Ah ok, nice. You should update that case statement to include the azurestack cluster profile.